components/nifosg/nifloader.cpp

```diff
@@ -1332,6 +1329,28 @@ namespace NifOsg
                             texEnv->setSource0_RGB(GL_PREVIOUS_ARB);
                             texEnv->setSource1_RGB(GL_TEXTURE);
                             stateset->setTextureAttributeAndModes(texUnit, texEnv, osg::StateAttribute::ON);
+                            break;
+                        }
+                        case Nif::NiTexturingProperty::GlossTexture:
+                        {
+                            std::cerr << "NiTexturingProperty::GlossTexture in " << mFilename << " not currently used." << std::endl;
+                            continue;
+                        }
+                        case Nif::NiTexturingProperty::BumpTexture:
+                        {
+                            std::cerr << "NiTexturingProperty::BumpTexture in " << mFilename << " not currently used." << std::endl;
+                            continue;
```

Comment:
You want to continue; before handling the texture. As it is now, e.g. a bump map would show up as a color texture, when it shouldn't show up at all so long as we can't handle it.